### DISCUSSION TITLE
Why do we need Actions if we can mutate the store directly from components / composition functions?

### DISCUSSION THREAD

**Discussion by martinszeltins** [vuejs/pinia#1276] [Help and Questions]
So with `Pinia` we can actually mutate the store directly from components or composition functions or from outside module. So what is the point of Actions now?

I mean I can create those "action" functions outside of the store, right...? or am I missing something here?

**Comment by posva** [comment]
An action holds the business logic related to the store similar to methods in components

**Comment by cohlar** [comment]
@martinszeltins I am wondering exactly the same thing.

One advantage I am currently experiencing in using composition functions over actions is that you can call other composition functions within your functions (for example UI success/error notifications), which I think you don't have access to within a store

I'm curious to know what are the pros of using actions over composition functions.

**Reply by martinszeltins** [reply]
@cohlar I have actually decided not to use actions inside a store and instead all my business logic lives inside services. That way there's a nice separation of concerns and my stores are only responsible for holding the data and computed getters.
